tests/simulation/test_initial_states.py

```diff
@@ -1,66 +1,130 @@
 """Tests for initial states conversion and validation utilities."""
 
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import TYPE_CHECKING
+
 import jax.numpy as jnp
 import pytest
 
+from lcm import DiscreteGrid, LinspaceGrid, Model, Regime
+from lcm.exceptions import InvalidInitialStatesError
 from lcm.simulation.util import (
     convert_flat_to_nested_initial_states,
     validate_flat_initial_states,
 )
-from tests.test_models.utils import get_model
-
 
-def test_convert_flat_to_nested_single_regime():
+if TYPE_CHECKING:
+    from lcm.typing import ContinuousState, DiscreteState, FloatND, ScalarInt
+
+
+@pytest.fixture
+def model() -> Model:
+    """Minimal model with two states (wealth, health) for initial states tests."""
+
+    @dataclass
+    class HealthStatus:
+        healthy: int = 0
+        sick: int = 1
+
+    @dataclass
+    class RegimeID:
+        active: int = 0
+        terminal: int = 1
+
+    def utility(wealth: ContinuousState, health: DiscreteState) -> FloatND:  # noqa: ARG001
+        return jnp.array(0.0)
+
+    def next_regime(period: int) -> ScalarInt:
+        return jnp.where(
+            period + 1 >= 2,
+            RegimeID.terminal,
+            RegimeID.active,
+        )
+
+    alive = Regime(
+        name="active",
+        utility=utility,
+        states={
+            "wealth": LinspaceGrid(start=1, stop=100, n_points=10),
+            "health": DiscreteGrid(HealthStatus),
+        },
+        transitions={
+            "next_wealth": lambda wealth: wealth,
+            "next_health": lambda health: health,
+            "next_regime": next_regime,
+        },
+    )
+
+    dead = Regime(
+        name="terminal",
+        terminal=True,
+        utility=lambda wealth, health: jnp.array([0.0]),  # noqa: ARG005
+        states={
+            "wealth": LinspaceGrid(start=1, stop=100, n_points=2),
+            "health": DiscreteGrid(HealthStatus),
+        },
+    )
+
+    return Model(
+        [alive, dead],
+        n_periods=2,
+        regime_id_cls=RegimeID,
+    )
+
+
+# ==============================================================================
+# Tests
+# ==============================================================================
+def test_convert_flat_to_nested_single_regime(model: Model) -> None:
```

Comment:
Name seems confusing now that we only ever have models with min 2 regimes. *(unless you think that supporting the static case as a special one is worth it, but then this would definitely be a misnomer)*